File: Difficulty: Medium/Count Subarrays with given XOR/count-subarrays-with-given-xor.cpp
class Solution {
  public:
    long subarrayXor(vector<int> &arr, int k) {
        // code here
        
        int n = arr.size();
        long ans=0;
        int xr = 0 ;
        unordered_map<int,int>m;
        m[0]++;
        for(int i = 0 ; i< n ; i++){
            
            xr^=arr[i];
            int find = xr^k;
            if(m.find(find)!=m.end()){
                ans+=m[find];
            }
            
            m[xr]++;
        }
        
        return ans;
    }
};

File: Difficulty: Medium/Count Subarrays with given XOR/count-subarrays-with-given-xor.py
class Solution:
    def subarrayXor(self, arr, k):
        # code here
        
        m={}
        m[0]=1
        
        ans=0
        xor=0
        for x in arr:
            
            xor^=x
            find = k^xor
            
            if find in m :
                ans+=m[find]
            
            m[xor]=m.get(xor,0)+1
        
        
        return ans
            
            
            